apps/main/src/dex/components/PageCrvLocker/Page.tsx

```diff
@@ -5,24 +5,25 @@ import styled from 'styled-components'
 import FormCrvLocker from '@/dex/components/PageCrvLocker/index'
 import type { FormType } from '@/dex/components/PageCrvLocker/types'
 import { ROUTE } from '@/dex/constants'
-import usePageOnMount from '@/dex/hooks/usePageOnMount'
 import Settings from '@/dex/layout/default/Settings'
 import useStore from '@/dex/store/useStore'
 import { type CrvLockerUrlParams, CurveApi } from '@/dex/types/main.types'
-import { getPath } from '@/dex/utils/utilsRouter'
+import { getPath, useChainId } from '@/dex/utils/utilsRouter'
 import Box, { BoxHeader } from '@ui/Box'
 import IconButton from '@ui/IconButton'
 import Spinner, { SpinnerWrapper } from '@ui/Spinner'
+import { isLoading, useConnection } from '@ui-kit/features/connect-wallet'
 import { t } from '@ui-kit/lib/i18n'
-import { useApiStore } from '@ui-kit/shared/useApiStore'
 
-const Page = (params: CrvLockerUrlParams) => {
+export const PageCrvLocker = (params: CrvLockerUrlParams) => {
   const { push } = useRouter()
-  const { routerParams, curve } = usePageOnMount()
-  const { rChainId, rFormType } = routerParams
-
+  const { lib: curve = null, connectState } = useConnection<CurveApi>()
+  const pageLoaded = !isLoading(connectState)
+  const rChainId = useChainId(params.network)
+  const {
```

Comment:
Last time this was used it caused issues as rFormType might be uninitialized, which breaks when destucturing. Hope that ain't the case here

Commit that fixed the issue: 7ca27aad6fe522e19096a8e53e1ef8dda02d2a33

Reply:
This is not safe to do anywhere, but this is directly in the page component, the props passed are derived from the route